c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2146,7 +2728,24 @@ class DeviceCachingAllocator {
     return true;
   }
 
+  void release_expandable_segment(Block* block) {
+    TORCH_INTERNAL_ASSERT(
```

Comment:
So at this point, there can only be a single block that points to a expandable segment, that's why the sizes must be equal?